azure-mgmt-storage/src/main/java/com/microsoft/azure/management/storage/implementation/CheckNameAvailabilityResult.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.storage.implementation;
+
+
+import com.microsoft.azure.management.storage.implementation.api.Reason;
+
+/**
+ * The CheckNameAvailability operation response.
+ */
+public class CheckNameAvailabilityResult {
+    /**
+     * Gets a boolean value that indicates whether the name is available for
+     * you to use. If true, the name is available. If false, the name has
+     * already been taken or invalid and cannot be used.
+     */
+    private Boolean nameAvailable;
```

Comment:
I don't think we want to use those classes representing primitives.
it should really be `boolean`, not `Boolean`